### PR TITLE
fix(DB/smart_scripts): Quest "Enlistment Day"

### LINKED ISSUE #3699: Valiance Keep recruiting event not scripted/Quest Dialogue [$30]
##### CURRENT BEHAVIOUR:

Recruitment Officer Carven and the Civilian Recruits are simply static and do not perform their continuous event.  The Civilian Recruits do not say anything when you accept the quest Enlistment Day (ID: 11672)

No dialogue between the 3 NPCs when you turn in the quest Enlistment Day (ID: 11672)

It's not scripted but most of it is found in the database, maybe a wrong trigger?

https://i.imgur.com/j3Kxm3T.jpg

##### EXPECTED BLIZZLIKE BEHAVIOUR:

There is an event that happens with the recruitment line, the transcript of which is found here: https://wow.gamepedia.com/Civilian_Recruit

At random, Recruitment Officer Carven will say one of these:

```
    Recruitment Officer Carven says: Yes, then. What is your trade?
    Recruitment Officer Carven says: State your profession.
    Recruitment Officer Carven says: Your profession?
    Recruitment Officer Carven says: Your previous line of work, recruit?
    Recruitment Officer Carven says: What did you do before you came to Northrend, then?
```

A civilian recruit will randomly respond with one of these:

```
    Civilian Recruit says: Cook.
    Civilian Recruit says: Farmhand.
    Civilian Recruit says: Farmhand, sir.
    Civilian Recruit says: Farmhand, m'lord.
    Civilian Recruit says: Miner.
    Civilian Recruit says: Miner, sir.
    Civilian Recruit says: Shipwright.
    Civilian Recruit says: Shipwright, sir.
    Civilian Recruit says: Mason, sir.
    Civilian Recruit says: Blacksmith.
    Civilian Recruit says: Blacksmith, sir.
    Civilian Recruit says: Carpenter.
    Civilian Recruit says: Carpenter, sir.
    Civilian Recruit says: Seamstress, m'lord.
    Civilian Recruit says: Tailor.

```

Recruitment Officer Carven will randomly respond with one of these:

```
    Recruitment Officer Carven says: Yeah. We could use some more of you. Report to the civilian liaison.
    Recruitment Officer Carven says: Looks like we have room for one more... no, my mistake. That's a seven not a one. Hope you're good with a sword - report to the barracks.
    Recruitment Officer Carven says: I thought we had enough in your line of work, but it looks like we have some openings. Report to the civilian liaison.
    Recruitment Officer Carven says: Yeah. We could use some more of you. Report to the civilian liaison for work assignment.
    Recruitment Officer Carven says: Afraid not, friend. Here's your sword and shield. Report to the barracks for duty.
    Recruitment Officer Carven says: Not anymore! Here's your sword. Report to the barracks for duty!
    Recruitment Officer Carven says: With a sword arm like that? I'm afraid we need you in the front lines, my friend. Report to the barracks.
    Recruitment Officer Carven says: I do have some openings in your line of work. Report to the civilian liaison.
    Recruitment Officer Carven says: You're breaking my heart - I've quotas to fill, you know? Can you wield a sword? Off to the barracks.
    Recruitment Officer Carven says: Yes, you're well seasoned in your field of work. Report to the civilian liaison at once, we need more like you!
```

When a Civilian Recruit leaves the recruitment line, one of these are said:
```

    Recruitment Officer Carven says: Keep it moving, people.
    Recruitment Officer Carven says: Next, please!
    Recruitment Officer Carven says: Next!
```

The Civilian Recruit will then leave the line, and the line moves up - a new Civilian Recruit spawns near the docks and gets in the back of the line.

When you accept the quest Enlistment Day (ID: 11672), a Civilian Recruit should say this:
`What's the matter, <class>? Think you're too good to stand in line with the rest of us?`

When you complete Enlistment Day, the following should be said:

```
    Harbinger Vurenn says: Give the word, general. I will have two regiments at your disposal in a month's time.
    Counselor Talbot says: Our troops, general, consist mostly of villagers and peasants. Good men, but not quite rid of the prejudices and superstitions of their upbringing. They're not ready to fight alongside our more exotic allies.
    General Arlos says: Your offer will be carefully considered, harbinger. In the meantime we will make use of your delegation in an advisory role.
    Harbinger Vurenn says: Very well, general. Should you change your mind, my people will be more than willing to provide military assistance.
```

https://wow.gamepedia.com/Enlistment_Day

##### STEPS TO REPRODUCE THE PROBLEM:

1. Create character, level doesn't matter
2. .tele valiancekeep
3. Walk to the recruitment line, there is no scripted event
4. Add the quest Enlistment Day with .quest add 11672 and turn it in to General Arlos in the barracks, there is no dialogue

##### AC HASH/COMMIT:

543753b641bc

##### OPERATING SYSTEM:

Win10

<bountysource-plugin>

---
There is a **[$30 open bounty](https://www.bountysource.com/issues/94324987-valiance-keep-recruiting-event-not-scripted-quest-dialogue?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by FALL1N1**
## CHANGES PROPOSED:
-  
-  

## ISSUES ADDRESSED:
- Closes #3699

## TESTS PERFORMED:
- http://prntscr.com/vlzfnw
-

## KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Voxstrasza** [comment]
thank you fallin! you're amazing, I tested this out asap and here's my feedback:

This happens upon server start/loading the map (empty space): https://i.imgur.com/mi9JzhI.jpg - but when new ones spawn and get into the line, they move up to begin the scripted event, and the NPCs in the 'static' line (the one that isn't supposed to move) aren't supposed to have weapons in their hand

Jumping on table: https://i.imgur.com/MTMRZaF.jpg, can this be moved back a bit so the NPCs don't jump on the table?

No dialogue after turning in Enlistment Day: https://i.imgur.com/OQWMBS0.jpg - I did get dialogue after accepting the quest though! (first screenshot)

**Comment by Voxstrasza** [comment]
Worldserver errors:

```
2020-11-19 14:52:52 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 1 for talker 25317, ignored.
2020-11-19 14:52:54 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 1 for talker 25317, ignored.
2020-11-19 14:52:58 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 10 using non-existent Text id 2 for talker 25317, ignored.
2020-11-19 14:53:00 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 10 using non-existent Text id 2 for talker 25317, ignored.
2020-11-19 14:53:04 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 3 for talker 25317, ignored.
2020-11-19 14:53:06 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 3 for talker 25317, ignored.
2020-11-19 14:53:30 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 1 for talker 25317, ignored.
2020-11-19 14:53:34 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 1 for talker 25317, ignored.
2020-11-19 14:53:36 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 10 using non-existent Text id 2 for talker 25317, ignored.
2020-11-19 14:53:40 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 10 using non-existent Text id 2 for talker 25317, ignored.
2020-11-19 14:53:42 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 3 for talker 25317, ignored.
2020-11-19 14:53:46 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 3 for talker 25317, ignored.
```

**Comment by FALL1N1** [comment]
Looks like we are missing texts, I'll re-check it tomorrow, thank you!

**Review by Si1ker — APPROVED** [review]

**Comment by Maurowin** [comment]
Do you will adds all dial for every location ?

**Comment by FALL1N1** [comment]
> 
> 
> Do you will adds all dial for every location ?

what do you mean?

**Comment by FALL1N1** [comment]
Added misc texts
Fixed the waypoint being on the table instead of in front of it (they still walk over a bit, but that's from sniffs, idk if it's ok to replace with custom WP while it'd indeed be better)
The NPCS are initially missing, that's blizzlike as of the SAI timers. Once there's a player around the event will automatically start and they will end up in a queue.

**Comment by Maurowin** [comment]
> 
> 
> > Do you will adds all dial for every location ?
> 
> what do you mean?

add all language, ES, FR, DE etc... for all dialogue

**Comment by FALL1N1** [comment]
No, just English, I do not have sniffs as I am backporting this from Trinity.

**Comment by Voxstrasza** [comment]
> 
> 
> Added misc texts
> Fixed the waypoint being on the table instead of in front of it (they still walk over a bit, but that's from sniffs, idk if it's ok to replace with custom WP while it'd indeed be better)
> The NPCS are initially missing, that's blizzlike as of the SAI timers. Once there's a player around the event will automatically start and they will end up in a queue.

https://i.imgur.com/H5AxUaC.jpg / https://i.imgur.com/konH2dX.jpg - up to you if you want to change it, I think it looks good.

https://i.imgur.com/GGGkqEU.jpg - works, civilian recruit practicing sword in the basement talks too much though

great work!! 💯

**Comment by FALL1N1** [comment]
@Voxstrasza it's fine by me, although it has to be approved by a team member.

File: data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql [hunk 1]
```diff
@@ -0,0 +1,144 @@
+DELETE FROM `creature_text` WHERE `CreatureID` in(25317,25220,25222);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql:1]: please use uppercase `IN` and add a space after it

File: data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+(25220, 3, 2277.823, 5238.724, 11.45096, 'Civilian Recruit'),
+(25220, 4, 2279.217, 5241.407, 11.45096, 'Civilian Recruit'),
+(25220, 5, 2280.84, 5244.218, 11.45719, 'Civilian Recruit'),
+(25220, 6, 2282.603, 5245.736, 11.36353, 'Civilian Recruit'),
+(25220, 7, 2284.867, 5246.299, 11.45096, 'Civilian Recruit'),
+(25220, 8, 2287.469, 5245.929, 11.45096, 'Civilian Recruit'),
+(25220, 9, 2289.469, 5244.898, 11.45096, 'Civilian Recruit'),
+(25220, 10, 2291.772, 5243.933, 11.45096, 'Civilian Recruit'),
+(25220, 11, 2294.129, 5242.708, 11.45096, 'Civilian Recruit'),
+-- (25220, 12, 2298.062, 5241.932, 12.3176, 'Civilian Recruit'),
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql:43]: ?

File: data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+(25220, 13, 2303.019, 5253.306, 11.50584, 'Civilian Recruit'),
+(25220, 14, 2308.73, 5256.926, 11.50584, 'Civilian Recruit'),
+(25220, 15, 2320.826, 5259.258, 11.25584, 'Civilian Recruit');
+
+DELETE FROM  `creature` WHERE `guid` IN(108008,108007,108006,108005,108004,108003,108002,108001,108000);
+
+UPDATE `creature_template` SET `AIName`='SmartAI' WHERE  `entry`IN(24959,25220,25307);
+DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid`IN(-107575,-107574);
+DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid`IN(25220,25307);
+DELETE FROM `smart_scripts` WHERE `source_type`=9 AND `entryorguid`IN(2522000,2522001);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql:54]: space after `IN`

File: data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+INSERT INTO `creature_addon` (`guid`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `auras`) VALUES 
+(117789, 0, 0, 3, 0, 0, NULL);
+
+-- Civilian Recruit SAI
+SET @GUID := -117794;
+UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=25317;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@GUID AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@GUID,0,0,0,1,0,100,0,8000,8000,12000,12000,5,36,0,0,0,0,0,1,0,0,0,0,0,0,0,"Civilian Recruit - Out of Combat - Play Emote 36"),
+(@GUID,0,1,0,1,0,100,0,35000,40000,35000,40000,80,11779400,0,0,0,0,0,1,0,0,0,0,0,0,0,"Civilian Recruit - Out of Combat - Run Script");
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql:101]: please use single quotes for strings

**Review by FrancescoBorzi — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql (hunk 2 above)
Comment (FALL1N1) [reply data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql:43]: Left for reference, the commented out point12 is blizzlike (they end up standing on the table), the custom point 12 is a bit back, so they don't do that. I have left a comment.

**Comment by Voxstrasza** [comment]
can the swords be removed on the Civilian Recruits in the line that doesn't move, just to make it a bit more blizzlike?

**Comment by Voxstrasza** [comment]
@FALL1N1 is this done?

**Comment by FALL1N1** [comment]
@Voxstrasza yes, it is.

**Comment by FALL1N1** [comment]
merged with master, updated and good to go

**Comment by Voxstrasza** [comment]
all good on my end!

**Comment by Helias** [comment]
Thanks for the test!

So, @Voxstrasza from your point of view we can put this as "To Be Merged"?

**Comment by Voxstrasza** [comment]
yes it works beautifully

File: data/sql/updates/pending_db_world/rev_1607432767438370200.sql [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+(25220, 12, 2297.330, 5241.240, 11.46370, 'Civilian Recruit'),
+(25220, 13, 2303.019, 5253.306, 11.50584, 'Civilian Recruit'),
+(25220, 14, 2308.73, 5256.926, 11.50584, 'Civilian Recruit'),
+(25220, 15, 2320.826, 5259.258, 11.25584, 'Civilian Recruit');
+
+DELETE FROM  `creature` WHERE `guid` IN (108008,108007,108006,108005,108004,108003,108002,108001,108000);
+
+UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry` IN (24959,25220,25307);
+DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid` IN (-107575,-107574);
+DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid` IN (25220,25307);
```
Comment (Si1ker) [inline data/sql/updates/pending_db_world/rev_1607432767438370200.sql:55]: @FALL1N1 ?
```suggestion
DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid` IN (-107575 ,-107574, 25220, 25307);
```

File: data/sql/updates/pending_db_world/rev_1607432767438370200.sql [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+(25317, 2, 0, 'You aren\'t going to let me sleep are you?', 12, 7, 100, 0, 0, 0, 26497, 'Civilian Recruit'),
+(25317, 2, 1, 'Are you going to be doing this for much longer?', 12, 7, 100, 0, 0, 0, 26502, 'Civilian Recruit'),
+(25317, 2, 2, 'Could you try to be a little quieter?', 12, 7, 100, 0, 0, 0, 26500, 'Civilian Recruit'),
+-- reaction 2
+(25317, 3, 0, 'I just don\'t want to let anyone down.', 12, 7, 100, 0, 0, 0, 26494, 'Civilian Recruit'),
+(25317, 3, 1, 'I need to do something to keep myself busy.', 12, 7, 100, 0, 0, 0, 26495, 'Civilian Recruit'),
+(25317, 3, 2, 'I\'m just nervous.  Sorry.', 12, 7, 100, 0, 0, 0, 26493, 'Civilian Recruit');
+
+DELETE FROM `creature_equip_template` WHERE `CreatureID` IN (25317, 25220);
+UPDATE `creature` SET `equipment_id`=0 WHERE `id` = 25317;
```
Comment (Si1ker) [inline data/sql/updates/pending_db_world/rev_1607432767438370200.sql:149]: new line at the end
```suggestion
UPDATE `creature` SET `equipment_id`=0 WHERE `id` = 25317;

```

File: data/sql/updates/pending_db_world/rev_1607432767438370200.sql [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
```
Comment (Si1ker) [inline data/sql/updates/pending_db_world/rev_1607432767438370200.sql:148]:
```suggestion
UPDATE `creature` SET `equipment_id`=0 WHERE `id` = 25317;


```

**Review by Si1ker — APPROVED** [review]

**Pull request merged**